### PR TITLE
Use different docker image for each base development branch

### PR DESCRIPTION
**Pull request by kmaehashi**
This fixes the issue that CuPy master is used for Chainer v6 tests.

After this got merged, configure CuPy webhook to trigger https://ci.preferred.jp/chainer.docker.py25and35.push/ and https://ci.preferred.jp/chainer.docker.py37.push/ for `v6` branch.

### CODE REVIEW

File: .pfnci/script.sh [hunk 1]
```diff
@@ -139,6 +148,26 @@ prepare_docker() {
...
+      return 0
+    fi
+  done
+  return 1
+}
+
+# get_base_branch returns the base development branch for the current HEAD.
+get_base_branch() {
+  for BASE_BRANCH in master v6; do
+    run git merge-base --is-ancestor "${BASE_BRANCH}" HEAD && echo "${BASE_BRANCH}" && return 0
```
Comment (take-cheeze) [inline .pfnci/script.sh]: I'm not sure but I can't run above command so maybe we need this:
```suggestion
    run git merge-base --is-ancestor "origin/${BASE_BRANCH}" HEAD && echo "${BASE_BRANCH}" && return 0
```

File: .pfnci/script.sh [hunk 2]
```diff
@@ -55,7 +58,7 @@ main() {
         docker_args+=(--env="SPREADSHEET_ID=${SPREADSHEET_ID}")
       fi
       run "${docker_args[@]}" \
-          "asia.gcr.io/pfn-public-ci/chainer-ci-prep.${TARGET}" \
+          "asia.gcr.io/pfn-public-ci/chainer-ci-prep.${TARGET}:${base_branch}" \
```
Comment (take-cheeze) [inline .pfnci/script.sh]: Are these docker tag going to be added afterward?

Comment (kmaehashi) [reply .pfnci/script.sh]: This docker image will be added via https://github.com/chainer/chainer/pull/8350/files#diff-8fe44c4bd2506d0a5db474fe567dd289R98.

Comment (take-cheeze) [reply .pfnci/script.sh]: OK

**Comment by kmaehashi** [comment]
Finally it's working!

**Comment by take-cheeze** [comment]
Jenkins, test this please!

**Comment by take-cheeze** [comment]
Jenkins, test this please!

**Comment by take-cheeze** [comment]
Jenkins, test this please!

**Pull request merged**